pkg/apply/taskrunner/task_test.go

```diff
@@ -4,98 +4,541 @@
 package taskrunner
```

Comment:
Note: this file has so many changes that it's effectively a rewrite. It's probably easier to review the result, rather than the diff.